redfish-core/lib/assembly.hpp

```diff
@@ -321,6 +368,80 @@ inline void setAssemblylocationIndicators(
         {
             setLocationIndicatorActive(asyncResp, assembly, iter->second);
         }
+
+        // Handle special case for tod_battery assembly OEM ReadyToRemove property
+        // NOTE: The following method for the special case of the tod_battery
+        // ReadyToRemove property only works when there is only ONE adcsensor
+        // handled by the adcsensor application.
+        if (sdbusplus::message::object_path(assembly).filename() == "tod_battery")
+        {
+            if (!json_util::readJson(req, asyncResp->res, "Assemblies", Assemblies, "Oem", oem))
+            {
+                return;
```

Comment:
Should have like a propertyValueFormatError

Reply:
This is following the logic of the following OEM feature of account_services.hpp
See: https://github.com/ibm-openbmc/bmcweb/blob/f3aa8d46eeaf72a517997f029dc42e0b60053334/redfish-core/lib/account_service.hpp#L2133
Can you be more specific of how messages::propertyValueFormatError should be used here?

Reply:
messages::propertyValueFormatError(
                asyncResp->res,
                thisJson.dump(2, ' ', true,
                              nlohmann::json::error_handler_t::replace),
                "Assemblies");

Reply:
To be clear, add propertyValueFormatError here

Reply:
There is no function called "thisJson" in BMCWEB.

Reply:
thisJson is just a nlohmann::json 
nlohmann::json has a dump
https://json.nlohmann.me/api/basic_json/dump/